### ISSUE TITLE
CSV Export: Modal redesign

### ISSUE THREAD

**Issue by Memo-Es** [opencollective/opencollective#7233]
## Overview

We are updating the experience to export the CSV file from the transactions section of the Fiscal Host dashboard...
![image](https://github.com/opencollective/opencollective/assets/49539954/33081c12-62cd-44cc-831a-6a8f4ce8021b)

The option to select a custom set of export fields is stressful for end-users:

1. There are many fields and it can feel overwhelming to review them all as they are currently presented.
2. Some fields are unclear and potentially confusing to users (eg: short ID’s and long ID’s?, amount & netAmount, isRefund, is Refunded), etc.).
3. There is no clarifying documentation.
4. Fields are not clearly organized/grouped.
5. We need to be able to add more fields and the form is already very long.
6. If a user does make an effort to select fields there is no way to save their selection for reuse in the future.

## Solution
Scope 1: Improvements to the existing UI

1. Better organize the fields on the screen (fields order, groups).
2. Adding tooltips to the fields.
3. Adding platform export presets (selected sets of fields) that enable us to provide multiple pre-saved configurations: eg: 
  1. “Accounting 2023” would represent our current set 
  2. “Accounting 2024” could be a modified version to account for the change in payment processor fees and taxes).
  3. and perhaps in the future other useful sets. 
4. Enable fiscal host admins to select output sets and save them as a fiscal-host set that all admins of the same fiscal host can use when exporting transactions.

Scope 2: Improved field selection

Enable a grouped-fields UI where:

* The groups themselves provide guidance and context.
* Focusing on group shows a limited set of fields (”hiding” all the other fields from view) that have shared context and are easier to navigate and select.
* The set of selected fields is clearly displayed and managed (remove fields).
* Optional:  change field order

**Comment by Memo-Es** [comment]
## Design update

After the first two team workshops, this is the current state of the design proposal:

https://www.loom.com/share/b69da52b69e24a3f9ec79109ec5adb06?sid=04f57439-51e0-43a2-abdb-8ea74166868e

@iamronen @znarf @kewitz

**Comment by kewitz** [comment]
Coda reference: https://coda.io/d/Make-Open-Collective_dnHLKv7oLV0/Currently_suFhs#CurrentProjects_tu75U/r145&view=modal

**Comment by kewitz** [comment]
Todo:
- [ ] Check if there's any necessary UI component missing
- [ ] Check existing implementations of reordering fields

**Comment by Memo-Es** [comment]
@kewitz About the pending design input:

**1. Placement of the separate as columns switch**
 - Below the "Selected fields for export"
![Image](https://github.com/opencollective/opencollective/assets/49539954/c9e5bc06-3bbf-4c9d-8a13-0a751cce4d91)

**2. Exporting action feedback**
 - [Loading state of the Export button](https://d.pr/i/6ovzbm)
 - Reference from https://ui.shadcn.com/docs/components/button

**3. File is too large**
 - Lock the modal to take any action until different filters are set.
![Image](https://github.com/opencollective/opencollective/assets/49539954/b14de481-88e1-4546-a931-4774f85cc49b)

[Figma file link](https://www.figma.com/file/ZQBMWhnGGtRWeIZknFW1eP/%5BOC-Design%5D-Production-Ready-%E2%9C%85?type=design&node-id=23363%3A20676&mode=design&t=YTDRF7zAzOqHT9ee-1)

**Comment by iamronen** [comment]
attn @Memo-Es copy for the backwards compatibility option:
- Delete the first part: Separate transactions compatibility
- Leave only: Export taxes and payment processor fees as columns.
- Add tooltip: Before 2024 payment processor fees and taxes were columns in transaction records. Since January  2024 they are separate transactions. Enable this option to transform separate payment processor fees and tax transactions into  columns in the export.

+ implementing the proposal from @kewitz to turn this on by default when selecting the 2023 platform preset (with the ability to turn it off).

**Comment by kewitz** [comment]
> [Loading state of the Export button](https://d.pr/i/6ovzbm)
> Reference from https://ui.shadcn.com/docs/components/button

Just a reminder that this export button is, in fact, a link that opens the download in a new tab. It does not export or generate anything.

**Comment by Memo-Es** [comment]
> Just a reminder that this export button is, in fact, a link that opens the download in a new tab. It does not export or generate anything.

Yes, the loading state is to have visual feedback on our platform parallel to the browser download

**Comment by Memo-Es** [comment]
@kewitz ok I see. Then we could use a toast notification saying something like: "Your export has started"
